packages/task/src/browser/task-service.ts

```diff
@@ -408,12 +454,184 @@ export class TaskService implements TaskConfigurationClient {
+                    // It may be a string (a task label) or a JSON object which represents a TaskIdentifier (e.g. {"type":"npm", "script":"script1"})
+                    const taskIdentifier = task.dependsOn[i];
+                    const dependentTask = this.getDependentTask(taskIdentifier, tasks);
+                    const taskCustomization = await this.getTaskCustomization(dependentTask);
+                    const resolvedMatchers = await this.resolveProblemMatchers(dependentTask, taskCustomization);
+                    dependentTasks.push({ 'task': dependentTask, 'taskCustomization': taskCustomization, 'resolvedMatchers': resolvedMatchers });
+                    // In case the 'dependsOrder' is 'sequence'
+                    if (task.dependsOrder && task.dependsOrder === 'sequence') {
+                        await this.runTasksGraph(dependentTask, tasks, {
+                            customization: { ...taskCustomization, ...{ problemMatcher: resolvedMatchers } }
+                        });
+                    }
+                }
+                // In case the 'dependsOrder' is 'parallel'
+                if (((!task.dependsOrder) || (task.dependsOrder && task.dependsOrder === 'parallel'))) {
+                    const promises = dependentTasks.map(item =>
+                        this.runTasksGraph(item.task, tasks, {
+                            customization: { ...item.taskCustomization, ...{ problemMatcher: item.resolvedMatchers } }
+                        })
+                    );
+                    await Promise.all(promises);
+                }
+            } else if (!Array.isArray(task.dependsOn)) {
+                // In case it is a string (a task label) or a JSON object which represents a TaskIdentifier (e.g. {"type":"npm", "script":"script1"})
+                const taskIdentifier = task.dependsOn;
+                const dependentTask = this.getDependentTask(taskIdentifier, tasks);
+                const taskCustomization = await this.getTaskCustomization(dependentTask);
+                const resolvedMatchers = await this.resolveProblemMatchers(dependentTask, taskCustomization);
+                await this.runTasksGraph(dependentTask, tasks, {
+                    customization: { ...taskCustomization, ...{ problemMatcher: resolvedMatchers } }
+                });
+            }
+        }
+
+        const taskInfo = await this.runTask(task, option);
+        if (taskInfo) {
+            const getExitCodePromise: Promise<TaskEndedInfo> = this.getExitCode(taskInfo.taskId).then(result => ({ taskEndedtype: TasksEndedTypes.TaskExited, value: result }));
+            const isBackgroundTaskEndedPromise: Promise<TaskEndedInfo> = this.isBackgroundTaskEnded(taskInfo.taskId).then(result =>
+                ({ taskEndedtype: TasksEndedTypes.BackgroundTaskEnded, value: result }));
+
+            // After start running the task, we wait for the task proccess to exit and if it is a background task, we also wait for a feedback
```

Comment:
[minor] typo - `proccess` should be `process`.

Reply:
fixed, done